### DISCUSSION TITLE
Multiple LSN support for single connection

### DISCUSSION THREAD

**Discussion by karolisg** [getdozer/dozer#1117] [Feature Requests]
At the moment we support single tuple of `(lsn, seq_no)`. It works fine when connector has one lsn for all tables. Altough, it doesn't work for `snowflake`, because there we have independent table streams and with single lsn tuple we cannot determine where to resume replication. @chubei has suggested this approach

> We can change the message identifier definition from per source to per source per output port, so we can still use one `ConnectorSource` but output independent message identifiers to different output ports and `Source::start_from` receives `HashMap<PortHandle, Option<(u64, u64)>` instead of `Option<u64, u64>`. Postgres connector is just a special case who always output same identifier to all ports, which the pipeline don't need to handle.

@snork-alt @v3g42 What do you think about such change?

**Comment by v3g42** [comment]
As discussed, makes sense to initialize separate connectors in my view instead of changing the connector interface. 

Maybe we can think of a way to share `connector pool` as a separate item ?
